synapse/handlers/oidc.py

```diff
@@ -246,6 +253,78 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
 
         await oidc_provider.handle_oidc_callback(request, session_data, code)
 
+    async def handle_backchannel_logout(self, request: SynapseRequest) -> None:
+        """Handle an incoming request to /_synapse/client/oidc/backchannel_logout
+
+        This extracts the logout_token from the request and try to figure out
```

Comment:
Maybe it is simple enough, but I wonder if we should have a separate method which parses the logout token -> `iss` & `aud` claims. Maybe overkill though if we don't need to d that anywhere else.

Reply:
It might make sense to refractor that if/when we implement RISC events (#11514) at some point since we'd have to do the same kind of matching, but I'd leave that refactor to when we do that if you don't mind

Reply:
That's fair, it ended up being simpler than I initially feared!